Symbol not found: _$s8RawValueSYTl with sentry-cli 2.50.1

### CLI Version

2.250.1

### Operating System and Architecture

- [x] macOS (arm64)
- [ ] macOS (x86_64)
- [ ] Linux (i686)
- [ ] Linux (x86_64)
- [ ] Linux (armv7)
- [ ] Linux (aarch64)
- [ ] Windows (i686)
- [ ] Windows (x86_64)

### Operating System Version

macos-13 Azure Pipelines

### Link to reproduction repository

_No response_

### CLI Command

sentry-cli login --auth-token $(sentry.auth.token)

### Exact Reproduction Steps

Appears to have just happened with a new build and seems to have occured with 2.50.1 that *just* came out. 

Our brew install step shows 2.50.1 being installed.

`🍺  /usr/local/Cellar/sentry-cli/2.50.1: 4 files, 24.0MB, built in 3 seconds`


Might well be AzurePipeline issue with their mac build agent, but not sure.

This is the full output of our cli command call,

```
/bin/bash --noprofile --norc /Users/runner/work/_temp/f5853b34-4906-4887-a8e7-a2b601d83b04.sh
dyld[6158]: Symbol not found: _$s8RawValueSYTl
  Referenced from: <633FE77A-3FFB-31AA-9349-F259F83B47F7> /usr/local/Cellar/sentry-cli/2.50.1/bin/sentry-cli
  Expected in:     <no uuid> unknown
/Users/runner/work/_temp/f5853b34-4906-4887-a8e7-a2b601d83b04.sh: line 1:  6158 Abort trap: 6           sentry-cli login --auth-token ***

##[error]Bash exited with code '134'.
```




### Expected Results

Login command to execute as before.

### Actual Results

Error: dyld[6158]: Symbol not found: _$s8RawValueSYTl


### Logs

```
dyld[6158]: Symbol not found: _$s8RawValueSYTl
  Referenced from: <633FE77A-3FFB-31AA-9349-F259F83B47F7> /usr/local/Cellar/sentry-cli/2.50.1/bin/sentry-cli
  Expected in:     <no uuid> unknown
/Users/runner/work/_temp/f5853b34-4906-4887-a8e7-a2b601d83b04.sh: line 1:  6158 Abort trap: 6           sentry-cli login --auth-token ***
```

--- a/.github/workflows/lint.yml
+++ b/.github/workflows/lint.yml
@@ -16,6 +16,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-24.04, macos-14, windows-2022]
+        feature-args: ['', '-Funstable-mobile-app']
         include:
           - os: ubuntu-24.04
             display-os: Linux
@@ -23,8 +24,12 @@ jobs:
             display-os: macOS
           - os: windows-2022
             display-os: Windows
+          - feature-args: ''
+            feature-suffix: ''
+          - feature-args: '-Funstable-mobile-app'
+            feature-suffix: ' (-Funstable-mobile-app)'
 
-    name: ${{ matrix.display-os }}
+    name: ${{ matrix.display-os }}${{ matrix.feature-suffix }}
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout Repository
@@ -40,4 +45,4 @@ jobs:
         run: cargo fmt --all -- --check
 
       - name: Run Clippy
-        run: cargo clippy --workspace --tests
+        run: cargo clippy --workspace --tests ${{ matrix.feature-args }}

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -16,6 +16,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ubuntu-24.04, macos-14, windows-2022]
+        feature-args: ['', '-Funstable-mobile-app']
         include:
           - os: ubuntu-24.04
             display-os: Linux
@@ -23,8 +24,12 @@ jobs:
             display-os: macOS
           - os: windows-2022
             display-os: Windows
+          - feature-args: ''
+            feature-suffix: ''
+          - feature-args: '-Funstable-mobile-app'
+            feature-suffix: ' (-Funstable-mobile-app)'
 
-    name: ${{ matrix.display-os }}
+    name: ${{ matrix.display-os }}${{ matrix.feature-suffix }}
     runs-on: ${{ matrix.os }}
 
     steps:
@@ -35,4 +40,4 @@ jobs:
         uses: swatinem/rust-cache@98c8021b550208e191a6a3145459bfc9fb29c4c0 # 2.8.0
 
       - name: Run Cargo Tests
-        run: cargo test --workspace
+        run: cargo test --workspace ${{ matrix.feature-args }}

--- a/.vscode/settings.json
+++ b/.vscode/settings.json
@@ -1,0 +1,4 @@
+{
+    "rust-analyzer.cargo.features": ["unstable-mobile-app"],
+    "rust-analyzer.cargo.noDefaultFeatures": false
+}

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,6 +11,7 @@ rust-version = "1.86"
 [dependencies]
 anylog = "0.6.3"
 anyhow = { version = "1.0.69", features = ["backtrace"] }
+apple-catalog-parsing = { path = "apple-catalog-parsing", optional = true }
 backoff = "0.4.0"
 brotli2 = "0.3.2"
 bytecount = "0.6.3"
@@ -93,6 +94,10 @@ default = []
 managed = []
 with_crash_reporting = []
 
+# Feature flag for the mobile-app command, as it is still under development.
+# CI tests run against this flag, but we don't include it in release builds.
+unstable-mobile-app = ["apple-catalog-parsing"]
+
 [workspace.lints.clippy]
 allow-attributes = "warn"
 str-to-string = "warn"
@@ -111,7 +116,6 @@ workspace = true
 
 [target."cfg(target_os = \"macos\")".dependencies]
 mac-process-info = "0.2.0"
-apple-catalog-parsing = { path = "apple-catalog-parsing"}
 
 [target."cfg(unix)"]
 

--- a/src/api/data_types/chunking/mobile_app.rs
+++ b/src/api/data_types/chunking/mobile_app.rs
@@ -1,3 +1,4 @@
+#![cfg(feature = "unstable-mobile-app")]
 use serde::{Deserialize, Serialize};
 use sha1_smol::Digest;
 

--- a/src/api/data_types/chunking/mod.rs
+++ b/src/api/data_types/chunking/mod.rs
@@ -14,5 +14,6 @@ pub use self::compression::ChunkCompression;
 pub use self::dif::{AssembleDifsRequest, AssembleDifsResponse, ChunkedDifRequest};
 pub use self::file_state::ChunkedFileState;
 pub use self::hash_algorithm::ChunkHashAlgorithm;
+#[cfg(feature = "unstable-mobile-app")]
 pub use self::mobile_app::{AssembleMobileAppResponse, ChunkedMobileAppRequest};
 pub use self::upload::{ChunkServerOptions, ChunkUploadCapability};

--- a/src/api/mod.rs
+++ b/src/api/mod.rs
@@ -1017,6 +1017,7 @@ impl<'a> AuthenticatedApi<'a> {
             .convert_rnf(ApiErrorKind::ReleaseNotFound)
     }
 
+    #[cfg(feature = "unstable-mobile-app")]
     pub fn assemble_mobile_app(
         &self,
         org: &str,

--- a/src/commands/mobile_app/mod.rs
+++ b/src/commands/mobile_app/mod.rs
@@ -1,3 +1,5 @@
+#![cfg(feature = "unstable-mobile-app")]
+
 use anyhow::Result;
 use clap::{ArgMatches, Command};
 

--- a/src/commands/mod.rs
+++ b/src/commands/mod.rs
@@ -57,6 +57,7 @@ macro_rules! each_subcommand {
         $mac!(info);
         $mac!(issues);
         $mac!(login);
+        #[cfg(feature = "unstable-mobile-app")]
         $mac!(mobile_app);
         $mac!(monitors);
         $mac!(organizations);

--- a/src/utils/mobile_app/mod.rs
+++ b/src/utils/mobile_app/mod.rs
@@ -1,3 +1,5 @@
+#![cfg(feature = "unstable-mobile-app")]
+
 #[cfg(target_os = "macos")]
 mod apple;
 mod validation;

--- a/tests/integration/mobile_app/mod.rs
+++ b/tests/integration/mobile_app/mod.rs
@@ -1,3 +1,5 @@
+#![cfg(feature = "unstable-mobile-app")]
+
 use crate::integration::TestManager;
 
 mod upload;

